lib/CFRPageActions.jsm

```diff
@@ -0,0 +1,225 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const POPUP_NOTIFICATION_ID = "contextual-feature-recommendation";
+
+/**
+ * A WeakMap from browsers to {host, recommendation} pairs. Recommendations are
+ * defined in the ExtensionDoorhanger.schema.json.
+ *
+ * A recommendation is specific to a browser and host and is active until the
+ * given browser is closed or the user navigates (within that browser) away from
+ * the host.
+ */
+const RecommendationMap = new WeakMap();
```

Comment:
it might be worth making these properties of `CFRPageActions` for future testing purposes

Reply:
I haven't actually moved the definitions otherwise there are circular no-use-before-defined issues but I added references to them to `CFRPageActions`